components/sync_manager/src/syncmanager.udl

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+namespace syncmanager { };
+
+[Error]
+enum SyncManagerError {
+    "UnknownEngine",
+    "UnsupportedFeature",
+    "ProtobufDecodeError",
+    "Sync15Error",
+    "UrlParseError",
+    "InterruptedError",
+    "JsonError",
+    "LoginsError",
+    "PlacesError",
+    "AnyhowError",
+};
+
+dictionary SyncParams {
+    SyncReason reason;
+    SyncEngineSelection engines;
+    record<DOMString, boolean> enabled_changes;
+    record<DOMString, string> local_encryption_keys;
+    SyncAuthInfo auth_info;
+    string? persisted_state;
+    DeviceSettings device_settings;
+};
+
+[Enum]
+interface SyncEngineSelection {
+    All();
+    Some(sequence<string> engines);
+};
+
+enum SyncReason {
+    "Scheduled",
+    "User",
+    "PreSleep",
+    "Startup",
+    "EnabledChange",
+};
+
+dictionary SyncAuthInfo {
+    string kid;
+    string fxa_access_token;
+    string sync_key;
+    string tokenserver_url;
+};
+
+dictionary DeviceSettings {
+    string fxa_device_id;
+    string name;
+    DeviceType kind;
+};
+
+enum DeviceType {
+    "Desktop",
+    "Mobile",
+    "Tablet",
+    "VR",
+    "TV",
+};
+
+dictionary SyncResult {
+    ServiceStatus status;
+    record<DOMString, string> failures;
+    sequence<string> successful;
+    string persisted_state;
+    sequence<string>? declined;
+    timestamp? next_sync_allowed_at;
+    string? telemetry_json;
```

Comment:
I know, so close.  I think having external types would also work, since then this could just be a `SyncTelemetryPing` record.  I also thought about wrapping this class in a kt file that we define and giving it a `telemetry()` method, but it just didn't seem worth it yet.